### PR TITLE
Originating desk not selected in the publishing dialog

### PR DESCRIPTION
**Pull request by dzonidoo**
SDESK-7398

### CODE REVIEW

File: scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts [hunk 1]
```diff
@@ -85,12 +85,13 @@ export function getInitialDestination(
     return destination;
 }
 
-export function getCurrentDeskDestination(): ISendToDestination {
-    let destinationDesk: string = sdApi.desks.getCurrentDeskId();
+export function getCurrentDeskDestination(id?: string): ISendToDestination {
+    const destinationDeskId: string = sdApi.desks.getCurrentDeskId();
+    const itemDeskId: string = sdApi.desks.getAllDesks().get(id)._id;
```
Comment (tomaskikutis) [inline scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts]: what is the purpose to call `getAllDesks`?

Comment (dzonidoo) [reply scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts]: change it to:
desk: id ?? destinationDeskId,

Comment (tomaskikutis) [reply scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts]: More than changing I'm interested why did you go with that approach to begin with. But yeah, I don't make sense get the entire object if you're not going to use anything from it. Unless you wanted to verify that desk ID exists, but then you should provide a fallback.

File: scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts [hunk 2]
```diff
@@ -85,12 +85,12 @@ export function getInitialDestination(
     return destination;
 }
 
-export function getCurrentDeskDestination(): ISendToDestination {
-    let destinationDesk: string = sdApi.desks.getCurrentDeskId();
+export function getCurrentDeskDestination(id?: string): ISendToDestination {
+    const destinationDeskId: string = sdApi.desks.getCurrentDeskId();
```
Comment (tomaskikutis) [inline scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts]: Can you remove this variable and inline this code?

Comment (dzonidoo) [reply scripts/core/interactive-article-actions-panel/utils/get-initial-destination.ts]: okay

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**